File: app/src/main/java/org/dhis2/usescases/about/AboutModule.java
package org.dhis2.usescases.about;

import androidx.annotation.NonNull;

import org.dhis2.data.dagger.PerFragment;
import org.dhis2.data.user.UserRepository;
import org.hisp.dhis.android.core.D2;

import dagger.Module;
import dagger.Provides;

/**
 * QUADRAM. Created by ppajuelo on 05/07/2018.
 */
@Module
public class AboutModule {

    @Provides
    @PerFragment
    AboutContracts.AboutPresenter providesPresenter(@NonNull D2 d2, @NonNull UserRepository userRepository) {
        return new AboutPresenterImpl(d2, userRepository);
    }
}

File: app/src/main/java/org/dhis2/usescases/about/AboutModule.kt
package org.dhis2.usescases.about

import dagger.Module
import dagger.Provides
import org.dhis2.commons.di.dagger.PerFragment
import org.dhis2.commons.schedulers.SchedulerProvider
import org.dhis2.data.user.UserRepository
import org.hisp.dhis.android.core.D2

@Module
class AboutModule(val view: AboutView) {
    @Provides
    @PerFragment
    fun providesPresenter(
        d2: D2,
        provider: SchedulerProvider,
        userRepository: UserRepository,
    ): AboutPresenter {
        return AboutPresenter(view, d2, provider, userRepository)
    }
}
